x-pack/filebeat/input/entityanalytics/provider/azuread/fetcher/graph/graph.go

```diff
@@ -353,21 +354,21 @@ func New(cfg *config.C, logger *logp.Logger, auth authenticator.Authenticator) (
 	if err != nil {
 		return nil, fmt.Errorf("invalid groups URL endpoint: %w", err)
 	}
-	groupsURL.RawQuery = url.QueryEscape(formatQuery(c.Select.GroupQuery, defaultGroupsQuery))
+	groupsURL.RawQuery = queryName + "=" + url.QueryEscape(formatQuery(c.Select.GroupQuery, defaultGroupsQuery))
```

Comment:
Should the `$select` be subjected to query escaping as well? IIUC the only part that we don't want to escape is the `=`. 

Perhaps building this up with `url.Values{...}.Encode()` would be safer since it handles the escaping for us.

Reply:
OK